### PR TITLE
Vulkan parity pass on the RTX 5080: the resident chain at 0.98x of llama.cpp b10660 on the 35B's prefill and 1.01x on its decode - the K-quant scale cache, the deltanet chain, the fused rail, the lens-derived readonly, and the review round's batch

### PR DESCRIPTION
**Pull request by borisbat**
**Behavior change: the Vulkan tier's prefill and decode kernels change their summation orders - a resident-vs-CPU logits bar reads a different noise sample than before; every parity file holds on its bar.**

**Why.** On a Linux RTX 5080 the resident chain ran the Qwen3.6-35B-A3B at 0.735x of llama.cpp b10660's pp512 (3833 against 5217 tok/s) and its decode at 0.91x; the dense hybrids sat at 0.76x to 0.89x on both.

**What changes.**
- The cm2 tiles' k step follows the format (32-deep on the five grid-codebook stamps), the k loop is hand-unrolled per superblock, and the decode callbacks of every K-quant and grid format are spelled in pair form on 16-bit lanes; the k4, k5 and iq4xs tiles stage a shared-memory scale cache.
- Split-k is decided by the dispatch group and sized by the role alone; the small f16 GEMMs of the router and the beta/alpha rows run on the cm2 tile with a split-k reduce, where the shape admits the route.
- The deltanet chain: the conv channel-major, the scan one column per lane cluster with the gates computed once per workgroup and the next token's row prefetched, the decode step column-per-thread in registers with the beta/alpha rows as one dispatch. The conv, the scan and the requant's K-quant fold map their 32-lane blocks from the invocation id, so the kernels hold on any subgroup width the tier admits.
- The `[vk_dispatch]` lens stamps NonWritable on every binding no kernel of a class writes; the fused add+rms+requant serves every site on every model (Q8_K and row-storing stamps, picked per site as `RqStamp`), the decode's z GEMV rides its own hazard class, the decode GEMV's block loop takes four blocks a lane straight-line.
- The GEMM probe gains the 9B shapes, a wave sweep, the decode head's isolation arm and two alternating output planes on every arm; the GEMV probe a single-plane arm; the decode profiler keys on recorded stamp names and says when its query pool saturates.

**Observable behavior.**
- Qwen3.6-35B-A3B UD-IQ2_XXS, Linux RTX 5080, pp512 3833 -> 5156 tok/s (0.983x of llama.cpp's 5247), tg32 132.5 -> 146.7 (1.010x of 145.2).
- Qwen3.5-9B UD-Q5_K_XL pp512 4369 -> 5145 (0.895x), tg32 91.7 -> 102.9 (0.912x); Qwen3.8-27B UD-IQ4_XS pp512 1496 -> 1711 (1.015x), tg32 39.3 -> 41.5; Qwen3.5-0.8B Q8_0 pp512 27524 -> 29016 (0.969x), tg32 336 -> 383.
- RTX 5060 Ti (the four-wide decode arm): the 35B pp512 2996 -> 3236 twin, the 9B tg32 unchanged within noise.
- `DASLLAMA_VK_FUSE=0` now pins every model's split add / norm / requant dispatches, not the Q8_0 dense case alone, and on a plain qk-norm model the split qk-norm/rope pair too; the tier announces the fused rail once at prepare.

**Where to look.** `dasllama_vulkan_classes.das` (the cm2 template's decode callbacks, the scale cache, `DnConv`, `DnScan`, `DnStepFused`, `ArRqT`, `gemv_shell`), `dasllama_vulkan_decode.das` (`rd_ensure_fused_sets`, `rq_stamp`, `rd_ts`), `dasllama_vulkan_dispatch.das` (`stamp_readonly`); the risky spots are the fused sites' per-consumer stamp pick, the hazard classes of the projection row's two halves, and the f16 GEMM route's shape gate (`f16g_shape_ok`).

<details>
<summary>Validation, claims, ledger</summary>

### Validation
- Every throughput figure is from the RunPod Linux RTX 5080 (driver 580.173.02, no decode-vector extension: the scalar decode arm), llama.cpp b10660 built there as the reference (`llama-bench -p 512 -n 32 -r 5`, its per-op figures from `GGML_VK_PERF_LOGGER=1`); CI has no such box. The pod is stopped.
- The kernel file (`tests/test_vulkan_kernels.das`): 122 of 122 on the pod's scalar arm, and 122 of 122 on this box's RTX 5060 Ti after the review batch (the twin arm; the earlier harvest run covered the scalar arm here too) - the cells include `test_vkd_f16_gemm`, `test_vkd_cm2e_batch`, `test_vkd_requant_family` (every stamp under a sentinel with a poisoned-expectation control), the deltanet family at head sizes 64 and 128, the scan cells at 1 row and the whole `DN_WINDOW`, and every `*_cm2_batch` cell's e and KHR arm lines (this box has KHR coopmat at subgroup 32, so the k4, k5 and iq4xs KHR arms ran).
- The module gate (`make-pr --only review-md`) reads OK on every discovered `REVIEW.das`; the image layout stamp is re-stamped (IMAGE_LAYOUT_STAMP_HASH 0x4098e7f964aefc1ful, IMAGE_VERSION 36 unchanged - the closure's `*_prepare` match reached the resident decode's prepare, see ASK below).
- Lint: 17 changed `.das` files clean on both rails (`utils/lint/main.das --quiet`, plain and `--disable-module dasMetal`). Formatter verified on the changed files.
- The walker test (`utils/internal/review-md/test_walkers.das`) 6 of 6; the spirv emitter cells 7 of 7; the whole `tests/spirv` suite 405 of 405 with spirv-val v2026.3 resolved, no leaks (`dastest -- --test tests/spirv --cov-path <scratch>/spirv.lcov --isolated-mode`).
- Model-free suite (`tests/run.das -- --suite model-free`, `-jit`, after the review batch): `REPORT suite model-free: 74 file(s), 0 failed, 4 cell(s) skipped (a filtered arm, an absent device, or a missing model or fixture - each skips loudly)`.
- Stocked suite, on the rebased tree with the rebuilt binary: all 47 files of `run.das`'s stocked list, run one at a time in the runner's exact per-child form (the same env, `--timeout`, `-jit -module-cache`) under `DASLLAMA_PARITY_FULL=1`, every file green - 46 in the sweep and `test_parity.das` rerun alone under the runner's new 3600 s budget (39 tests, 34 passed, 5 skipped, 1077 s). The one-at-a-time form was chosen deliberately: two back-to-back `--suite stocked` runs on this box ended in a display-driver death (a live kernel event 141 at 08:14:54 during `test_kquant.das`, then the freeze at 08:56), and the per-file sweep also counted the driver's events before and after each file - none moved across all 47. The runner fix in this PR: the stocked gate's children had the 1200 s arm-mode budget, which the large-tier parity file cannot meet (it ran 990-1334 s on this box), so the gate's parity file failed by construction on every run before; stocked children now get the full gate's 3600 s.
- The three resident parity files in that sweep: `test_gpu_resident_hybrid.das` 15 of 15, `test_gpu_resident_moe.das` 9 of 9 (the 2.7B, 35B and 30B fixtures), `test_gpu_resident_qwen2.das` 3 of 3; `test_gpu_moe_shexp.das` 3 of 3. The hybrid file's `DASLLAMA_VK_KV32=1` and `DASLLAMA_COOPMAT=mm` arms and the serving census were not rerun after the rebase (the driver death cut that chain); they ran green on the pre-batch tree, and the batch changed no route.
- Driver-health finding, not this PR's: the box logs GPU engine timeouts (live kernel event 141) during stocked runs - eleven distinct fault addresses today, none reproduced by any file alone. Detection is a follow-up in `followup_vulkan.md` item 45 (memory-budget sampling, `VK_EXT_device_fault` on device-lost, memory-priority pinning).
- Probe rows: the local RTX 5060 Ti sweep at the k-step commit ran `cm2:<fmt>` for all fourteen cm2 formats (both decode arms in one process each) plus `moe:iq2xxs`, `moesk:iq2xxs`, `moe:iq2s` and `moe:k4`, every arm exit 0; every later tile lever (the unroll, the pair-form decodes, the scale cache, the 9B shapes, the wave sweep, the decode head's isolation arm) carries its pod rows in `PERF_LEDGER.md` and `followup_vulkan.md` item 45 (`cm2:k4`/`k5`/`k6`/`iq4xs`, `moesk:iq2xxs`/`iq2s`/`iq3s`, `cm2d:k5`, `cm2w:`, `dec`). No post-batch re-sweep: the batch changed no tile arithmetic, and the kernel cells hold every format's l/m/s/e and KHR arms.
- The `#one-benchmark-rig` anchor grew by 57 lines; its 15 citers were audited: 14 match the section (every probe arm this branch added is described); `harness/p0_ko_probe.das:45`'s citation covers only its `tune_gate()` clause (pre-existing, unchanged here).
- `make-pr` on the rebased tip (sync, stamp-reach, jit-smoke, then `preflight --full` once, 1662 s): 18 gates passed, 4 skipped, 3 red - all three triaged, none a defect of this branch. The interpreter and JIT suites each report one item: `tests/watchdog/test_watchdog.das` at 79 s and 92 s against preflight's 60 s per-file cap; run alone on an idle box it passes (47 of 49, 2 skipped) in 81 s in both tiers, and this branch does not touch it - master's 2026-09-10 change to its deadlines is what put it over the cap. The utils-tests and AOT gates could not relink `libDaScriptDyn.dll` under preflight's own daslang host (the DLL pin the skill names); both targets were built from a plain shell afterwards and the gates rerun alone. utils-tests then ran its suites (531 s) with one red cell, `utils/daspkg/test_daspkg.das`'s "init_ansi_colors with --color flag", which fails whenever `NO_COLOR` is set in the environment - the detached shell the rerun was launched from carries `NO_COLOR=1`; the file rerun with the variable unset: 246 tests, 245 passed, 0 failed, 1 skipped. The full `test_aot` build (green from a plain shell, 5.5 min) and `dastest --use-aot` over `tests/`: 1485 files, 13332 tests, 13315 passed, 0 failed, 0 errors, 17 skipped.
- `make-pr --only dupes` (1268 candidates, 484 with matches): every exact cluster is the corpus's trivial-setter / one-line-getter shape (`set_x(v)` against every other setter) and the marker-builtin stubs of the emitter; the fuzzy matches sit under 0.86 across unrelated helpers - no missed reuse.
- The review round: grounding, six dimensions, 16 checklist auditors, the tdd-auditor (69 branches: 23 distinguished, 17 controlled, 26 untested, 3 unproven; 17 test edits justified), three style-hygiene auditors, seven surfacers, the prover, a woodpecker (codex) round pinned at 664ba9780, one dragon wave and one cold re-read - the re-read's serious findings landed in the closing batch (the two-pass cap is now written into the skills and the dragon charter).

### Claims - stated, not tested
- The lane-block form of the deltanet conv, the scan and the requant's K-quant fold (blocks from the invocation id, not the subgroup id) is correct by construction on a 64-lane subgroup; no 64-lane device is on hand, so the behavior there is untested. On the 32-lane cards here the kernels are identical to the subgroup-id form.
- The f16 router route (`F16GemmCm2` with the split-k reduce) and the fused rail's per-site picks were adopted on cross-commit before/after rows on the pod, not on a same-process race; the rows are in `followup_vulkan.md` item 45.
- No new end-to-end route: every model still takes the resident driver's default route; the levers move kernels inside it.
- The KHR (mm-mode) arm of the cm2 template keeps the scale cache's `@workgroup` array declared (unfilled) beside its own staging arrays; the emitter declares every `@workgroup` field, so the KHR stamp's declared footprint grows by 8320 B - measured by the khrx rows above, not by a KHR-only card.
- The emitter suite validates its fixtures; the engine's iq2xxs cm2 stamp modules still fail spirv-val's OpVariable placement check (a `[spirv_decode]` body's locals are not hoisted) - ledgered in item 45, the driver accepts them and the tiles match the oracle.

### ASK
- `REVIEW.das`'s image layout stamp closure hashes every function named `*_prepare` under `dasllama/`, so a resident-decode edit moves the stamp though no image layout moved - narrow the match (item 45, follow-up 3), or accept the re-stamps.
- The dragon's cold re-read proposed deleting the four `Weakening check_X ... is a defect` residues in `REVIEW_GPU_VULKAN.md` (lines 102, 104, 113, 149) as duplicates of the root checklist's `REVIEW.das` rule; left in place - a rule deletion is your call.
- `benchmarks/setup_lcpp_ref.das` pins `DEFAULT_REF_SHA` b10659 while this arc's reference figures are b10660 (master's pin, not this diff).
- `spawn_readonly_fixture` (the lens refusal gate's child compile) passes no `-dasroot`, so it audits the binary's own tree rather than the tree under test - item 45, follow-up 4.

### Lint candidates
- A `REVIEW.das` gate flagging `gl_SubgroupID` / `gl_SubgroupInvocationID` arithmetic against a literal lane count in a kernel the tier does not width-gate (the conv/scan/requant defect class).
- A gate flagging an `[unroll]` range or `@workgroup` array size that must track a named constant the kernel reads (`DN_CONV_PB`'s class).
- A `tests/spirv/REVIEW.das` gate for the roster-fixture correspondence (would retire two rules of `tests/spirv/REVIEW.md`).
- A `REVIEW.das` gate for the out-of-folder test ledger in `tests/CLAUDE.md`.

### Not done
- The dense hybrids' remaining decode gap (0.8B 0.80x, 27B 0.88x, 9B 0.91x) and the 9B's prefill (0.895x): the next pass, after the Vulkan coverage of the other families - the lever list with its expected gains is `followup_vulkan.md` item 45, its follow-ups list the review round's seven items (the nine duplicate e classes, the double convert, the stamp closure's match, the lens gate's `-dasroot`, an f16-vs-f32 router cell, the f16 route's work-size gate, the emitter's OpVariable placement).

</details>

### CODE REVIEW

File: tests/spirv/_spirv_common.das [hunk 1]
```diff
@@ -2827,3 +2912,26 @@ def ray_query {
 def public ray_query_words : array<uint> {
     return clone_to_move(ray_query_spv)
 }
+
+//! a module's OpConstant value by result id (its third operand); 0 when no such constant
```
[the remark this answers is not in the document]
Comment (borisbat) [reply tests/spirv/_spirv_common.das]: Accepted, fixed in 918fc69c5: the doc now says the result id is the instruction's second operand after the type and the value its third, which is what the code reads (w[opStart + 1], w[opStart + 2]).

File: tests/spirv/test_coopmat2_e.das [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+[test]
+def test_coopmat2_e_tiles(t : T?) {
+    t |> run("cm2 e-stamp shapes: A 128x32, B 32x128, acc f16 128x128, acc f32 128x128 through the name-parsed marker rail") <| @(t : T?) {
+        var words <- coopmat2e_words()
+        t |> success(length(words) >= 2 && words[1] == SPV_VERSION_1_6, "SPIR-V header version 1.6")
+        t |> success(op_has_operand(words, SpvOp.Capability, uint(SpvCapability.CooperativeMatrixKHR)),
+            "OpCapability CooperativeMatrixKHR")
+        var got <- spv_coopmat_shapes(words)
+        var want <- ["128x128", "128x128", "128x32", "32x128"]
+        t |> equal(want, got, "the OpTypeCooperativeMatrixKHR shapes (A, B, f16 acc, f32 acc) - the e stamp's, not the s stamp's")
```
[the remark this answers is not in the document]
Comment (borisbat) [reply tests/spirv/test_coopmat2_e.das]: Accepted, fixed in 918fc69c5: the message now names the compare as the sorted shape set - the e stamp's 128x32 and 32x128 operand tiles and its two 128x128 accumulators - with no operand order implied.

File: tests/spirv/test_coopmat2_s.das [hunk 3]
```diff
@@ -1,31 +1,43 @@
...
+require _spirv_common
+require spirv/spirv_dis
+require spirv/spirv_grammar
+
+//! the shape set is the claim: a fixture's four tile types at exactly these rows x cols, one MulAdd, the f16 -> f32 convert
+def private check_cm2_tiles(t : T?; words : array<uint>; lbl : string; want : array<string>) {
+    t |> success(length(words) >= 2 && words[1] == SPV_VERSION_1_6, "{lbl}: SPIR-V header version 1.6")
+    t |> success(op_has_operand(words, SpvOp.Capability, uint(SpvCapability.CooperativeMatrixKHR)), "{lbl}: OpCapability CooperativeMatrixKHR")
+    var got <- spv_coopmat_shapes(words)
+    t |> equal(want, got, "{lbl}: the OpTypeCooperativeMatrixKHR shapes (A, B, f16 acc, f32 acc)")
```
[the remark this answers is not in the document]
Comment (borisbat) [reply tests/spirv/test_coopmat2_s.das]: Accepted, fixed in 918fc69c5: the message now calls the compare the sorted shape set of the four tile types (the two operand tiles and the f16 and f32 accumulators), matching the helper's contract and the cell's own doc line.
